utopia-remix/app/services/fgaClient.server.ts

```diff
@@ -0,0 +1,65 @@
+import { CredentialsMethod, ErrorCode, FgaApiValidationError, OpenFgaClient } from '@openfga/sdk'
+import { ServerEnvironment } from '../env.server'
+
+const mockOpenFgaClient = {
+  write: async () => {},
+  check: async () => ({ allowed: true }),
+}
+
+export const fgaClient = createFgaClient()
+
+export function createFgaClient() {
+  const fgaClientMode = getFgaClientMode()
+  if (fgaClientMode === 'local_mock') {
+    return mockOpenFgaClient
+  }
+  return new OpenFgaClient({
+    apiScheme: ServerEnvironment.FGA_API_SCHEME,
+    apiHost: ServerEnvironment.FGA_API_HOST,
+    storeId: ServerEnvironment.FGA_STORE_ID,
+    // authorizationModelId: 'YOUR_MODEL_ID', // optional
+    credentials: getCredentials(fgaClientMode),
+  })
+}
+
+function getCredentials(fgaClientMode: 'local' | 'remote') {
+  // if working against a local server, we don't need credentials
+  if (fgaClientMode === 'local') {
+    return { method: CredentialsMethod.None } as const
+  } else {
+    return {
+      method: CredentialsMethod.ClientCredentials,
+      config: {
+        apiTokenIssuer: ServerEnvironment.FGA_API_TOKEN_ISSUER,
+        apiAudience: ServerEnvironment.FGA_API_AUDIENCE,
+        clientId: ServerEnvironment.FGA_CLIENT_ID,
+        clientSecret: ServerEnvironment.FGA_SECRET,
+      },
+    } as const
+  }
+}
+
+function getFgaClientMode(): 'local_mock' | 'local' | 'remote' {
+  if (ServerEnvironment.environment === 'local') {
+    if (ServerEnvironment.FGA_API_HOST == '') {
+      return 'local_mock'
+    }
+    if (ServerEnvironment.FGA_API_HOST.includes('localhost')) {
+      return 'local'
+    }
+  }
+  return 'remote'
+}
+
+export async function safeFgaWrite(params: object) {
+  try {
+    await fgaClient.write(params)
+  } catch (err) {
+    if (((err as FgaApiValidationError).apiErrorCode = ErrorCode.WriteFailedDueToInvalidInput)) {
```

Comment:
`===`!

Reply:
how could this pass the tests? are we using the same linter config as for the editor?

Reply:
Our disgusting way of running eslint has come back to haunt us!
The CI version only runs on the main editor: https://github.com/concrete-utopia/utopia/blob/master/editor/package.json#L18
And the commit hook doesn't include the new `utopia-remix` package: https://github.com/concrete-utopia/utopia/blob/master/editor/package.json#L59-L99

Reply:
yes, we saw that the remix config is not the same 🤦  it also doesn't warn on hooks dependencies array or let/const...